### PR TITLE
test: fix compilation problem on debian 8.

### PR DESCRIPTION
**Pull request by NickNaso**
Fixes https://github.com/nodejs/node-addon-api/issues/1137

### CODE REVIEW

**Comment by NickNaso** [comment]
First pass on CI:
| Version | Job | Status |
|---:|---|:---:|
| main | https://ci.nodejs.org/job/node-test-node-addon-api-new/5382/ |  |
| v17.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5383/ |  |
| v16.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5384/ |  |
| v14.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5385/ |  |
| v12.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5386/ |  |

**Comment by NickNaso** [comment]
Second pass on CI:
| Version | Job | Status |
|---:|---|:---:|
| main | https://ci.nodejs.org/job/node-test-node-addon-api-new/5399/ |  |
| v17.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5400/ |  |
| v16.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5401/ |  |
| v14.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5402/ |  |
| v12.x | https://ci.nodejs.org/job/node-test-node-addon-api-new/5403/ |  |

File: test/objectwrap_function.cc [hunk 1]
```diff
@@ -36,7 +36,8 @@ class FunctionTest : public Napi::ObjectWrap<FunctionTest> {
 };
 
 std::unordered_map<napi_env, Napi::FunctionReference>
-    FunctionTest::constructors = {};
+    FunctionTest::constructors =
+        std::unordered_map<napi_env, Napi::FunctionReference>();
```
Comment (addaleax) [inline test/objectwrap_function.cc:40]: I think just dropping the `=` or the ` = {}` would also do the trick here

**Review by addaleax — APPROVED** [review]

File: test/objectwrap_function.cc (hunk 1 above)
Comment (mhdawson) [reply test/objectwrap_function.cc:40]: That's what I just tested out, removing the = {}

File: test/objectwrap_function.cc [hunk 2]
```diff
@@ -36,7 +36,8 @@ class FunctionTest : public Napi::ObjectWrap<FunctionTest> {
 };
 
 std::unordered_map<napi_env, Napi::FunctionReference>
-    FunctionTest::constructors = {};
+    FunctionTest::constructors =
```
Comment (mhdawson) [inline test/objectwrap_function.cc:39]:
```suggestion
    FunctionTest::constructors;
```

File: test/objectwrap_function.cc [hunk 3]
```diff
@@ -36,7 +36,8 @@ class FunctionTest : public Napi::ObjectWrap<FunctionTest> {
 };
 
 std::unordered_map<napi_env, Napi::FunctionReference>
-    FunctionTest::constructors = {};
+    FunctionTest::constructors;
+        std::unordered_map<napi_env, Napi::FunctionReference>();
```
Comment (mhdawson) [inline test/objectwrap_function.cc:40]:
```suggestion
```

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Going to land as soon as the github actions complete

**Comment by mhdawson** [comment]
landed as e906b5a7ce248db3df6f1e856fda04d6dd14f6a7. @NickNaso thanks for the PR.

**Pull request closed without merging**